libraries/primitive_consul_health.rb

```diff
@@ -0,0 +1,54 @@
+require_relative 'primitive'
+require 'chef/mash'
+
+module Choregraphie
+  class ConsulHealthCheck < Primitive
+    def initialize(options = {})
+      @options = Mash.new(options)
+
+      @options[:tries] ||= 10
+      @options[:checkids] ||= []
+      @options[:delay] ||= 15
+
+      if @options[:consul_token]
+        require 'diplomat'
+        Diplomat.configure do |config|
+          config.acl_token = @options[:consul_token]
+        end
+      end
+
+      # this block could be used to configure diplomat if needed
+      yield if block_given?
+    end
+
+    def register(choregraphie)
+      choregraphie.cleanup do
+        require 'diplomat'
+        tries = @options[:tries]
+        all_passing = false
+        failing_checks = []
+
+        while !all_passing && tries != 0
+          failing_checks = []
+          checks = Diplomat::Check.checks
+
+          non_passing = @options[:checkids]
+            .select { |id| checks[id]['ID'] }
+            .reject { |id| checks[id]['Status'] == 'passing' }
+
+          all_passing = !non_passing.any?
+
+          non_passing.each { |id| Chef::Log.warn "Check #{id} failed" }
+
+          if all_passing
+            break
+          else
+            Chef::Log.warn "Some checks did not pass, will retry #{tries} more times"
+            tries -= 1
+            Kernel::sleep @options[:delay]
+          end
```

Comment:
:skull_and_crossbones:    behavior when we've done n "tries" is the same than passing tests. Don't you want to do something differently?